cmd/node/main.go

```diff
@@ -384,3 +394,40 @@ func prepareWbtWatcher(client k8sClient.Client, eventsRecorder *events.Recorder,
 
 	return wbt.NewConfWatcher(client, eventsRecorder, ll, nodeKernel), nil
 }
+
+// getDriverManagerType retrieves the type of the driver manager used by the pod.
+// It searches for a container named "drivemgr" in the specified pod and
+// determines the driver manager type based on its image.
+//
+// If the driver manager type is found, it returns the type as a string and nil error.
+// If the pod or container is not found, it returns an empty string and an error indicating
+// the failure reason.
+func getDriveManagerType(client k8sClient.Client, logger *logrus.Logger) (string, error) {
+	pod := corev1.Pod{}
+	if err := client.Get(context.Background(), k8sClient.ObjectKey{Name: os.Getenv("POD_NAME"), Namespace: *namespace}, &pod); err != nil {
+		return "unable to find pod", err
+	}
+	return findDriverManager(pod, logger)
+}
+
+func findDriverManager(pod corev1.Pod, logger *logrus.Logger) (string, error) {
+	for _, c := range pod.Spec.Containers {
```

Comment:
have you consider using a map  to store the associations between driver manager types and their corresponding image strings? it will simplify the code. Example from codeium:

 // Define a map to store the associations between driver manager types and their corresponding image strings
var driverManagerImages = map[string]string{
	v1.DriveManagerLoopback: v1.DriveManagerLoopbackImage,
	v1.DriveManagerHal:     v1.DriveManagerHalImage,
	v1.DriveManagerBase:    v1.DriveManagerBaseImage,
	v1.DriveManagerIDRAC:   v1.DriveManagerIDRACImage,
}

// ...

for _, c := range pod.Spec.Containers {
	if c.Name == "drivemgr" {
		logger.Debugf("found a driver manager's image: %s", c.Image)
		for driverManagerType, image := range driverManagerImages {
			if strings.Contains(c.Image, image) {
				return driverManagerType, nil
			}
		}
		logger.Errorf("unknown driver manager type for image: %s", c.Image)
	}
}

Reply:
Great point!